packages/vscode-extension/src/project/deviceSession.ts

```diff
@@ -87,6 +87,12 @@ export class DeviceSession implements Disposable {
           break;
       }
     });
+
+    // we create a and start debug session here to be able to leverage the functionality of debug console
+    // the session is not connected to the debugger here yet and that step can only happen,
+    // after app is running.
+    this.debugSession = new DebugSession(this.debugEventDelegate);
+    this.debugSession.startDebugSession();
```

Comment:
We should avoid exposing too many unnecessary implementation details. If this isn't useful to have separate control over creating and starting, the constructor should just start the session on its own.